src/ProjectService.Business/Commands/EditProjectCommand.cs

```diff
@@ -0,0 +1,113 @@
+using LT.DigitalOffice.ProjectService.Mappers.RequestsMappers.Interfaces;
+using LT.DigitalOffice.ProjectService.Models.Db;
+using LT.DigitalOffice.ProjectService.Models.Dto.Enums;
+using LT.DigitalOffice.ProjectService.Models.Dto.Requests;
+using LT.DigitalOffice.ProjectService.Models.Dto.Responses;
+using LT.DigitalOffice.ProjectService.Validation.Interfaces;
+using MassTransit;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace LT.DigitalOffice.ProjectService.Business.Commands
+{
+    public class EditProjectCommand : IEditProjectCommand
+    {
+        private readonly IEditProjectValidator _validator;
+        private readonly IAccessValidator _accessValidator;
+        private readonly IPatchDbProjectMapper _mapper;
+        private readonly IProjectRepository _repository;
+        private readonly IRequestClient<IGetDepartmentRequest> _requestClient;
+        private readonly ILogger<CreateProjectCommand> _logger;
+
+        private IGetDepartmentResponse GetDepartment(Guid departmentId, List<string> errors)
+        {
+            string errorMessage = "Cannot edit project. Please try again later.";
+
+            try
+            {
+                var response = _requestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>(
+                IGetDepartmentRequest.CreateObj(null, departmentId)).Result;
+
+                if (response.Message.IsSuccess)
+                {
+                    return response.Message.Body;
+                }
+
+                _logger.LogWarning($"Can not find department with this id '{departmentId}': " +
+                    $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}");
+            }
+            catch (Exception exc)
+            {
+                _logger.LogError(exc, errorMessage);
+
+                errors.Add(errorMessage);
+            }
+
+            return null;
+        }
+        public EditProjectCommand(
+            IEditProjectValidator validator,
+            IAccessValidator accessValidator,
+            IPatchDbProjectMapper mapper,
+            IProjectRepository repository,
+            IRequestClient<IGetDepartmentRequest> requestClient,
+            ILogger<CreateProjectCommand> logger
+        )
+        {
+            _validator = validator;
+            _accessValidator = accessValidator;
+            _mapper = mapper;
+            _repository = repository;
+            _requestClient = requestClient;
+            _logger = logger;
+        }
+
+        public OperationResultResponse<bool> Execute(Guid projectId, JsonPatchDocument<EditProjectRequest> request)
+        {
+            _validator.ValidateAndThrowCustom(request);
+
+            if (!_accessValidator.IsAdmin() && !_accessValidator.HasRights(Kernel.Constants.Rights.AddEditRemoveProjects))
```

Comment:
I think it is bad have rights to edit all projects

Reply:
in future this will correct

Reply:
hmm, why don't correct it now?

Reply:
done